libp2p/network/connection/raw_connection.py

```diff
@@ -52,4 +53,5 @@ def __init__(
 
     async def close(self) -> None:
         self.writer.close()
-        await self.writer.wait_closed()
+        if sys.version_info[0:2] > (3, 6):
```

Comment:
a bit clunky for the `wait_closed` API only in `3.7+` -- open to suggestions on how to handle this kind of thing...

Reply:
One feature I aim for is: the ability to simply delete lines when we drop support for an old python version.

Which makes me think of:
```py
if sys.version_info < (3, 7):  # one bonus of this construction is that you don't have to slice the version_info
  return
await self.writer.wait_closed()
```

I don't love early returns, but might personally favor that to get the "delete to drop support" feature.

Some other options:
1. subclass the writer with an empty `wait_closed()` if using python3.6 (should reduce the number of different files touched, which is nice)
2. basically the same as ^ but monkey-patch the method onto the `StreamWriter` instead of subclassing
3. a `hasattr(writer, 'wait_closed')` check first (tiny performance penalty, and doesn't meet the goal of "delete to drop support")

I personally have higher tolerance for "uglier" patch solutions here, since it will be deleted when we drop support for old versions. (Like 1 or 2 above)

Any of these (including status quo) is fine, just offering alternatives that you might like better.

Reply:
yeah i like the early return version actually; the goal of being able to just delete lines to drop support for older versions does seem like a worthwhile thing to aim for!